client.go

```diff
@@ -270,6 +275,47 @@ func (c *TunaSessionClient) shouldAcceptAddr(addr string) bool {
 	return false
 }
 
+func (c *TunaSessionClient) getPubAddrsFromRemote(remoteAddr string, config *nkn.DialConfig) (*PubAddrs, error) {
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	if config.DialTimeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(config.DialTimeout)*time.Millisecond)
+		defer cancel()
+	}
+
+	buf, err := json.Marshal(&Request{Action: "getPubAddr"})
+	if err != nil {
+		return nil, err
+	}
+
+	var msg *nkn.Message
+	for i := 0; i < 5; i++ {
+		respChan, err := c.multiClient.Send(nkn.NewStringArray(remoteAddr), buf, nil)
+		if err != nil {
+			return nil, err
+		}
+		select {
+		case msg = <-respChan.C:
+			break
+		case <-time.After(3 * time.Second):
```

Comment:
Shouldn't have this case if it's last try

Reply:
A better way is that, we create 3 goroutines, `i = 0, 1, 2`. For goroutine `i`:
- sleep (i * 3s) (with dial context)
- if still hasn't got any response (done chan not closed yet), send msg, otherwise return
- get response (with dial context)
- send response to resp chan, close done chan